### PR TITLE
Add MariaDB sql dialect

### PR DESCRIPTION
**Pull request by panilya**

### CODE REVIEW

File: src/main/java/net/datafaker/transformations/SqlDialect.java [hunk 1]
```diff
@@ -12,6 +12,7 @@ public enum SqlDialect {
     LUCIDDB("\""),
     MSSQL("[]"),
     MYSQL("`", Casing.UNCHANGED),
+    MARIADB("`"),
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/transformations/SqlDialect.java:15]: Please keep abc order

Comment (snuyanzin) [reply src/main/java/net/datafaker/transformations/SqlDialect.java:15]: I guess it should have same casing as Mysql or am I wrong?

Comment (panilya) [reply src/main/java/net/datafaker/transformations/SqlDialect.java:15]: This is why I'm requesting your review. The documentation says that it depends on OS. For Windows, it's not case-sensitive. For Linux, Database, table, table aliases and trigger names are affected by the systems case-sensitivity, while index, column, column aliases, stored routine and event names are never case sensitive.

Comment (snuyanzin) [reply src/main/java/net/datafaker/transformations/SqlDialect.java:15]: Do you have a link to documentation?

Comment (panilya) [reply src/main/java/net/datafaker/transformations/SqlDialect.java:15]: https://mariadb.com/kb/en/library/identifier-case-sensitivity/

Comment (snuyanzin) [reply src/main/java/net/datafaker/transformations/SqlDialect.java:15]: it is stated that
>The [lower_case_table_names](https://mariadb.com/kb/en/server-system-variables/#lower_case_table_names) server system variable plays a key role. It determines whether table names, aliases and database names are compared in a case-sensitive manner. If set to 0 (the default on Unix-based systems), table names and aliases and database names are compared in a case-sensitive manner. If set to 1 (the default on Windows), names are stored in lowercase and not compared in a case-sensitive manner. If set to 2 (the default on Mac OS X), names are stored as declared, but compared in lowercase.

it looks like in that case it is either lowercase or case sensitive => `Casing.To_LOWER` should be applied. Even if this parameter is not set on db side name will be quoted and it is ok

Comment (panilya) [reply src/main/java/net/datafaker/transformations/SqlDialect.java]: Fixed

**Review by snuyanzin — APPROVED** [review]

**Pull request merged**